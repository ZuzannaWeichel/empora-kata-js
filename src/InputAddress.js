const Lookup = require("smartystreets-javascript-sdk/src/us_street/Lookup");

class InputAddress extends Lookup{
  constructor(inputData){
    super()
    
    this.street = inputData["Street"],
    this.city = inputData["City"],
    this.zipCode = inputData["Zip Code"]
  }
}

module.exports = InputAddress