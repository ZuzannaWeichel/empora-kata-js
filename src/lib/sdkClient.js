import SmartySDK from "smartystreets-javascript-sdk";

const SmartyCore = SmartySDK.core;

const authId = process.env.SMARTY_AUTH_ID;
const authToken = process.env.SMARTY_AUTH_TOKEN;

const credentials = new SmartyCore.StaticCredentials(authId, authToken);
const clientBuilder = new SmartyCore.ClientBuilder(credentials);

export const sdkClient = clientBuilder.buildUsStreetApiClient();

