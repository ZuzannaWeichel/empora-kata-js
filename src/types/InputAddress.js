import Lookup from "smartystreets-javascript-sdk/src/us_street/Lookup.js";

export default class InputAddress extends Lookup{
  constructor(inputData){
    super()
    
    this.street = inputData["Street"],
    this.city = inputData["City"],
    this.zipCode = inputData["Zip Code"]
  }
}