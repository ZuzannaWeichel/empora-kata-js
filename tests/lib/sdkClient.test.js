import { sdkClient } from '../../src/lib/sdkClient.js';
import InputAddress from '../../src/types/InputAddress.js';

describe('sdkClient', () => {
  it('should be authenticated', () => {
    expect(process.env.SMARTY_AUTH_ID).toBeDefined();
    expect(process.env.SMARTY_AUTH_TOKEN).toBeDefined();
  });

  it('should return response', async () => {
    const lookup = new InputAddress({ Street: '143 e Maine Street', City: 'Columbus', 'Zip Code': '43215' });
 
    const response = await sdkClient.send(lookup);

    expect(response).toMatchSnapshot();
  });
});