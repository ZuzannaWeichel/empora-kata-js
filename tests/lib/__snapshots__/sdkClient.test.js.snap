// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sdkClient should return response 1`] = `
Batch {
  "lookups": [
    InputAddress {
      "addressee": undefined,
      "city": "Columbus",
      "inputId": undefined,
      "lastLine": undefined,
      "match": undefined,
      "maxCandidates": undefined,
      "result": [
        Candidate {
          "addressee": undefined,
          "analysis": {
            "active": "Y",
            "cmra": "N",
            "dpvFootnotes": "AAN1",
            "dpvMatchCode": "D",
            "enhancedMatch": undefined,
            "footnotes": "H#M#",
            "isEwsMatch": undefined,
            "isSuiteLinkMatch": undefined,
            "lacsLinkCode": undefined,
            "lacsLinkIndicator": undefined,
            "noStat": "Y",
            "vacant": "N",
          },
          "candidateIndex": 0,
          "components": {
            "cityName": "Columbus",
            "defaultCityName": "Columbus",
            "deliveryPoint": "99",
            "deliveryPointCheckDigit": "2",
            "extraSecondaryDesignator": undefined,
            "extraSecondaryNumber": undefined,
            "plus4Code": "5370",
            "pmbDesignator": undefined,
            "pmbNumber": undefined,
            "primaryNumber": "143",
            "secondaryDesignator": undefined,
            "secondaryNumber": undefined,
            "state": "OH",
            "streetName": "Main",
            "streetPostdirection": undefined,
            "streetPredirection": "E",
            "streetSuffix": "St",
            "urbanization": undefined,
            "zipCode": "43215",
          },
          "deliveryLine1": "143 E Main St",
          "deliveryLine2": undefined,
          "deliveryPointBarcode": "432155370992",
          "inputIndex": 0,
          "lastLine": "Columbus OH 43215-5370",
          "metadata": {
            "buildingDefaultIndicator": "Y",
            "carrierRoute": "C023",
            "congressionalDistrict": "03",
            "coordinateLicense": "SmartyStreets",
            "countyFips": "39049",
            "countyName": "Franklin",
            "elotSequence": "0237",
            "elotSort": "A",
            "isEwsMatch": undefined,
            "latitude": 39.9567,
            "longitude": -82.99384,
            "obeysDst": true,
            "precision": "Zip9",
            "rdi": "Commercial",
            "recordType": "H",
            "timeZone": "Eastern",
            "utcOffset": -5,
            "zipType": "Standard",
          },
          "smartyKey": undefined,
        },
      ],
      "secondary": undefined,
      "state": undefined,
      "street": "143 e Maine Street",
      "street2": undefined,
      "urbanization": undefined,
      "zipCode": "43215",
    },
  ],
}
`;
